vcd/resource_vcd_nsxt_distributed_firewall_test.go

```diff
@@ -0,0 +1,681 @@
+  description = "IP Set containing IPv4 and IPv6 ranges"
+
+  ip_addresses = [
+    "12.12.12.1",
+    "10.10.10.0/24",
+    "11.11.11.1-11.11.11.2",
+    "2001:db8::/48",
+    "2001:db6:0:0:0:0:0:0-2001:db6:0:ffff:ffff:ffff:ffff:ffff",
+  ]
+}
+
+resource "vcd_nsxt_ip_set" "set2" {
+  edge_gateway_id = vcd_nsxt_edgegateway.nsxt-edge.id
+
+  name        = "{{.Name}}-ipset2"
+  description = "Empty IP Set"
+}
+
+resource "vcd_nsxt_security_group" "g1-empty" {
+  edge_gateway_id = vcd_nsxt_edgegateway.nsxt-edge.id
+
+  name = "{{.Name}} empty group"
+}
+
+resource "vcd_nsxt_security_group" "g2" {
+  edge_gateway_id = vcd_nsxt_edgegateway.nsxt-edge.id
+
+  name = "{{.Name}} group with members"
+  member_org_network_ids = [vcd_network_routed_v2.nsxt-backed.id]
+}
+
+resource "vcd_network_routed_v2" "nsxt-backed" {
+  name            = "{{.Name}}-routed-net"
+  edge_gateway_id = vcd_nsxt_edgegateway.nsxt-edge.id
+
+  gateway       = "1.1.1.1"
+  prefix_length = 24
+
+  static_ip_pool {
+    start_address = "1.1.1.10"
+    end_address   = "1.1.1.20"
+  }
+}
+
+resource "vcd_nsxt_app_port_profile" "p1" {
+  org = "{{.Org}}"
+  context_id = vcd_vdc_group.test1.id
+  name = "{{.Name}}-app-profile"
+
+  scope = "TENANT"
+
+  app_port {
+    protocol = "ICMPv6"
+  }
+
+  app_port {
+    protocol = "TCP"
+    port     = ["2000", "2010-2020"]
+  }
+
+  app_port {
+    protocol = "UDP"
+    port     = ["40000-60000"]
+  }
+}
+
+data "vcd_nsxt_manager" "main" {
+  name = "{{.NsxtManager}}"
+}
+
+data "vcd_nsxt_app_port_profile" "WINS" {
+  context_id = data.vcd_nsxt_manager.main.id
+  name       = "WINS"
+  scope      = "SYSTEM"
+}
+
+data "vcd_nsxt_app_port_profile" "FTP" {
+  context_id = data.vcd_nsxt_manager.main.id
+  name       = "FTP"
+  scope      = "SYSTEM"
+}
+`
+
+const dfwStep2 = constDfwPrereqs + `
+resource "vcd_nsxt_distributed_firewall" "t1" {
+  org          = "{{.Org}}"
+  vdc_group_id = vcd_vdc_group.test1.id
+
+  rule {
+	name        = "rule1"
+	action      = "ALLOW"
+	description = "description"
+
+	source_ids           = [vcd_nsxt_ip_set.set1.id, vcd_nsxt_ip_set.set2.id]
+	destination_ids      = [vcd_nsxt_security_group.g1-empty.id, vcd_nsxt_security_group.g2.id]
+	app_port_profile_ids = [vcd_nsxt_app_port_profile.p1.id, data.vcd_nsxt_app_port_profile.WINS.id, data.vcd_nsxt_app_port_profile.FTP.id]
+  }
+
+  rule {
+	name      = "rule2"
+	action    = "DROP"
+	enabled   = false
+	logging   = true
+	direction = "IN_OUT"
+
+	source_ids                  = [vcd_nsxt_ip_set.set1.id, vcd_nsxt_ip_set.set2.id]
+	network_context_profile_ids = [
+		data.vcd_nsxt_network_context_profile.cp1.id,
+		data.vcd_nsxt_network_context_profile.cp2.id,
+		data.vcd_nsxt_network_context_profile.cp3.id
+	]
+  }
+
+  rule {
+	name        = "rule3"
+	action      = "DROP"
+	ip_protocol = "IPV4"
+  }
+
+  rule {
+	name        = "rule4"
+	action      = "ALLOW"
+	ip_protocol = "IPV6"
+	direction   = "OUT"
+  }
+
+  rule {
+	name        = "rule5"
```

Comment:
It would be possible but migrating all parameters will make it like 40+ parameters. That will make it a lot harder to read configuration (not saying it is very easy to read it even now, but that is the cost of having many params)